browser-extension/mv3/src/service-worker/services/apiClient/index.ts

```diff
@@ -89,6 +96,16 @@ export async function getAPIResponse(apiRequest: Request): Promise<Response | {
 
   try {
     const requestStartTime = performance.now();
+    let responseHeaders: KeyValuePair[] = [];
+    apiRequestCorrelationManager.addHandler(
+      requestlyId,
+      (requestDetails: chrome.webRequest.WebResponseHeadersDetails) => {
+        responseHeaders = requestDetails.responseHeaders.map((header) => ({
+          key: header.name,
+          value: header.value,
+        }));
+      }
```

Comment:
As this is async, is it possible that these are executed after the response is returned to the UI?

Reply:
`fetch` awaits for the response being returned. We listen for the response headers on `onHeadersReceived` which is triggered before any response is received. 

In [webRequest terminology](https://developer.chrome.com/docs/extensions/reference/api/webRequest#life_cycle_of_requests), we listen on `webRequest.onHeadersReceived` and fetch is completed when `webRequest.onCompleted` is triggered. This ensures that response headers are always available before the response is received.

We can also read response headers from fetch response as a fallback.

Reply:
Can you try adding a delay in headersReceived and see if what happens? 

AFAIK, now we are using non blocking APIs so this would return nothing

Reply:
Delay wont work. Its async.
We are just using the sequence of events because `fetch` call can be awaited